mipengine/common/node_catalogue.py

```diff
@@ -0,0 +1,44 @@
+import importlib.resources as pkg_resources
+import json
+from dataclasses import dataclass
+from typing import Dict, List, Any
+
+from dataclasses_json import dataclass_json
+
+from mipengine import resources
+from mipengine.controller.common.utils import Singleton
+
+
+@dataclass_json
+@dataclass
+class Node:
+    nodeId: str
+    rabbitmqURL: str
+    monetdbURL: str
+    data: Dict[str, List[str]]
+
+
+@dataclass_json
+@dataclass
+class NodeCatalogue(metaclass=Singleton):
+    global_node: Node
+    local_nodes: Dict[str, Node]
+    data: Dict[str, List[str]]
+
+    def __init__(self):
+        node_catalogue_str = pkg_resources.read_text(resources, 'node_catalogue.json')
```

Comment:
minor thing, but adding _str does not convey much information about the purpose of the variable. 
_filename on the other hand, shows that this holds the filename of the node catalog file, and it is thus probably a string